src/ucs/datastruct/bitmap.h

```diff
@@ -359,18 +363,60 @@ typedef uint64_t ucs_bitmap_word_t;
     }
 
 
+/**
+ * Find the index of the first bit set to 1 in a given bitmap, starting from
+ * a particular index (excluding it)
+ *
+ * @param bitmap      Look for the first bit in this bitmap
+ * @param nun_words   Number of words in the bitmsp
+ * @param start_index The first bit to look from
+ */
+int _ucs_bitmap_ffs(void *bitmap, size_t num_words, size_t start_index)
+{
+    size_t word_index        = start_index / UCS_BITMAP_BITS_IN_WORD;
+    size_t next_bit          = start_index;
+    ucs_bitmap_word_t *words = (ucs_bitmap_word_t*)bitmap;
+
+    if (start_index % UCS_BITMAP_BITS_IN_WORD) {
+        /* If start_index is not the last bit the word, filter out all bits
+           up to it and continue scanning this word */
+        next_bit = ucs_ffs64_safe(
+                words[word_index] &
+                ~UCS_MASK(start_index % UCS_BITMAP_BITS_IN_WORD));
+        if (next_bit != UCS_BITMAP_BITS_IN_WORD) {
+            /* A bit found in this word */
+            return _UCS_BITMAP_BIT_INDEX(next_bit, word_index);
+        } else {
+            word_index++;
+        }
+    }
+
+    /* Continue scanning next words, until a word with a set bit is found */
+    while (word_index < num_words) {
```

Comment:
can we unite the 2 cases?
1. initialize mask to ~UCS_MASK(start_index % UCS_BITMAP_BITS_IN_WORD) (no need to check start_index % UCS_BITMAP_BITS_IN_WORD)
2. if (word[index]&mask) return ucs_ffs64(words[word_index] & mask);
3. mask = 0xfff..fff
4. goto (2)

Reply:
Done.